clustermesh-apiserver/clustermesh/converters.go

```diff
@@ -101,3 +114,151 @@ func (ic *IdentityConverter) Convert(event resource.Event[*cilium_api_v2.CiliumI
 	kv := store.NewKVPair(identity.Name, string(lbls))
 	return singleIter[store.Key](kv), noneIter[store.NamedKey]
 }
+
+// ----- CiliumEndpoints - CiliumEndpointSlices ----- //
+
+func newEndpointOptions(cfg cmk8s.CiliumEndpointSliceConfig) Options[*types.CiliumEndpoint] {
+	return Options[*types.CiliumEndpoint]{
+		Enabled:   !cfg.EnableCiliumEndpointSlice,
+		Resource:  "CiliumEndpoint",
+		Prefix:    path.Join(ipcache.IPIdentitiesPath, ipcache.DefaultAddressSpace),
+		StoreOpts: []store.WSSOpt{store.WSSWithSyncedKeyOverride(ipcache.IPIdentitiesPath)},
+	}
+}
+
+// EndpointConverter implements Converter[*types.CiliumEndpoint]
+type EndpointConverter struct {
+	endpointConverterCommon[*types.CiliumEndpoint]
+}
+
+func newEndpointConverter(logger *slog.Logger, cinfo cmtypes.ClusterInfo) Converter[*types.CiliumEndpoint] {
+	return &endpointConverterCommon[*types.CiliumEndpoint]{
+		cache: make(map[string]sets.Set[string]),
+		iter:  (&EndpointConverter{}).all,
+	}
+}
+
+func newEndpointSliceOptions(cfg cmk8s.CiliumEndpointSliceConfig) Options[*cilium_api_v2a1.CiliumEndpointSlice] {
+	return Options[*cilium_api_v2a1.CiliumEndpointSlice]{
+		Enabled:   cfg.EnableCiliumEndpointSlice,
+		Resource:  "CiliumEndpointSlice",
+		Prefix:    path.Join(ipcache.IPIdentitiesPath, ipcache.DefaultAddressSpace),
+		StoreOpts: []store.WSSOpt{store.WSSWithSyncedKeyOverride(ipcache.IPIdentitiesPath)},
+	}
+}
+
+// EndpointSliceConverter implements Converter[*cilium_api_v2a1.CiliumEndpointSlice]
+type EndpointSliceConverter struct {
+	endpointConverterCommon[*cilium_api_v2a1.CiliumEndpointSlice]
+}
+
+type endpointConverterCommon[T interface {
+	runtime.Object
+	*types.CiliumEndpoint | *cilium_api_v2a1.CiliumEndpointSlice
+}] struct {
+	iter  func(T) iter.Seq2[string, identity.IPIdentityPair]
+	cache map[string]sets.Set[string]
```

Comment:
nit: maybe something like that would be more readable?
```suggestion
	ipcache map[resource.Key]sets.Set[string]
```

Reply:
Changed the map key to `resource.Key`, that's definitely much better. I've kept the cache name instead, both because now it serves a more general purpose, and because the `ipcache` term is fairly overloaded in Cilium.